### PR TITLE
feat: expose function to send minidumps

### LINKED ISSUE #1050: Allow capturing minidumps programatically
A customer who has their own crash handler mechanism would like to upload minidumps they capture by themselves.
One of the solutions proposed was using HTTP and [uploading them to our `minidump` endpoint](https://docs.sentry.io/platforms/native/guides/minidumps/).

Since that means having to deal with retries, managing files on disk etc. I suggesting using one of our SDKs to manage that.

Our envelope format allows attachment, including type `minidump`. Which should trigger symbolication.

While we might not have a `capture_minidump` yet, the suggestion is to create this function so the SDK takes over the job of establishing the connection to Sentry. And deleting the file upon completion.

### PR DESCRIPTION
**Pull request by vaind**
Adds `sentry_capture_minidump(const char*)` to send an existing minidump. 

This PR builds on top of #1052 but makes significant API changes so I've made a new PR instead of rewriting the old one.

* Closes #1050
* Closes #1052 

Note: CI failures seem to be unrelated, see https://github.com/getsentry/sentry-native/issues/1069

### CODE REVIEW

**Comment by bruno-garcia** [comment]
Thank you @vaind !

File: src/sentry_core.c [hunk 1]
```diff
@@ -18,6 +18,7 @@
 #include "sentry_tracing.h"
 #include "sentry_transport.h"
 #include "sentry_value.h"
+#include "transports/sentry_disk_transport.h"
```
Comment (supervacuus) [inline src/sentry_core.c:21]: This is no longer necessary, right? I would love to keep direct dependencies from core to concrete implementations at a minimum.

File: tests/test_capture_minidump.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+pytestmark = pytest.mark.skipif(not has_http, reason="tests need http")
+
+# fmt: off
+auth_header = (
+    "Sentry sentry_key=uiaeosnrtdy, sentry_version=7, sentry_client=sentry.native/0.7.11"
+)
+# fmt: on
+
+
+def test_capture_minidump(cmake, httpserver):
```
Comment (supervacuus) [inline tests/test_capture_minidump.py:23]: Is this function sufficiently different from all other HTTP integration tests to end up in a separate module?

File: include/sentry.h [hunk 3]
```diff
@@ -1362,6 +1362,12 @@ SENTRY_API sentry_user_consent_t sentry_user_consent_get(void);
  */
 SENTRY_API sentry_uuid_t sentry_capture_event(sentry_value_t event);
 
+/**
+ *  Allows capturing independently created minidumps.
```
Comment (supervacuus) [inline include/sentry.h:1366]: Given that this interface has no peers in the standard SDK interfaces, a quick spec of what happens is in order here, especially that:

* it generates a dummy event with log-level `FATAL` and
* this generated event is processed like any other event, including attachments, scope application, and before-send hook.

**Review by supervacuus — CHANGES_REQUESTED** [review]
Thanks! Only minor things.

File: include/sentry.h (hunk 3 above)
Comment (vaind) [reply include/sentry.h:1366]: OK, updated the doc

**Pull request merged**